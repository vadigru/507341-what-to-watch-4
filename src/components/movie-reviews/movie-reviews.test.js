import React from "react";
import renderer from "react-test-renderer";
import MovieReviews from "./movie-reviews.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import Namespace from "../../reducer/namespace.js";

const mockStore = configureStore([]);

const mock = [
  {
    id: 1,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

it(`Should render MovieDetails component`, () => {
  const store = mockStore({
    [Namespace.DATA]: {
      reviews: []
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <MovieReviews
            reviews={mock}
          />
        </Provider>
    )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
