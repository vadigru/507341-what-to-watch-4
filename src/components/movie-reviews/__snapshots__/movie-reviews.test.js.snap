// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render MovieDetails component 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment text.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            June 25, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment text.
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          />
          <time
            className="review__date"
            dateTime="2016-12-24"
          >
            June 25, 2020
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8.9
      </div>
    </div>
  </div>
</div>
`;
